src/main/scala/uk/gov/nationalarchives/tdr/api/http/Routes.scala

```diff
@@ -23,6 +25,29 @@ class Routes(val config: Config) extends Cors {
   val ttlSeconds: Int = 10
   val url: String = config.getString("auth.url")
 
+  def unmarshalEntity(entity: ResponseEntity): String = {
+    Unmarshal(entity).to[String].toString
+  }
+
+  def isNon200Status(status: StatusCode): Boolean = {
+    val code = status.intValue.toString
+    !code.startsWith("2")
+  }
+
+  def logging: Directive0 = {
+    def logNon200Response(req: HttpRequest)(res: RouteResult): Unit = {
+      res match {
+        case Complete(resp) =>
+          if (isNon200Status(resp.status)) {
+            val responseEntity = unmarshalEntity(resp.entity)
+            logger.info(s"Non 200 Response: $req\nStatus Code: ${resp.status}\nResponse Entity: $responseEntity")
```

Comment:
What do you think about logging the whole request body, which is probably going to be the key to debugging failed requests?

Reply:
Whole request body is logged.